nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/RedisDocumentMessageConsumer.java

```diff
@@ -0,0 +1,163 @@
+        String blobPath = "";
+        String blobFilename = "";
+        String blobMimeType = "";
+        String properties;
+        try {
+            if (blobInfo == null || StringUtils.isBlank(blobInfo.filename)) {
+                properties = getProperties(message);
+            } else {
+                blobPath = blobInfo.filename;
+                blobMimeType = blobInfo.mimeType;
+                blobFilename = Paths.get(blobInfo.filename).getFileName().toString();
+                properties = getPropertiesWithBlob(message);
+            }
+            String parentPath = message.getParentPath();
+            if (parentPath.startsWith("/")) {
+                parentPath = parentPath.substring(1);
+            }
+            String key = Paths.get(parentPath, message.getName()).toString();
+            String docKey = redisPrefix + ":" + DOC_KEY_SUFFIX;
+            String folderKey = redisPrefix + ":" + FOLDER_KEY_SUFFIX;
+            String dataKey = redisPrefix + ":" + DATA_KEY_SUFFIX + ":" + key;
+            String level = String.valueOf(key.split("/").length);
+            String url = URLEncoder.encode(key, "UTF-8").replaceAll("%2F", "/");
+            String payload = getPayload(message, properties);
+            redisExecutor.evalsha(addDocumentSHA, Arrays.asList(bytes(docKey), bytes(dataKey), bytes(folderKey)),
+                    Arrays.asList(bytes(key), bytes(parentPath), bytes(message.getType()), bytes(message.getName()),
+                            bytes(payload), bytes(url), bytes(level), bytes(blobPath), bytes(blobFilename),
+                            bytes(blobMimeType)));
+        } catch (UnsupportedEncodingException e) {
+            throw new NuxeoException("Cannot urlencode: " + message, e);
+        } catch (JsonProcessingException e) {
+            throw new NuxeoException("Cannot convert properties to json: " + message, e);
+        }
+    }
+
+    protected String getPayload(DocumentMessage message, String properties) {
+        return String.format("{\"entity-type\": \"document\", \"name\": \"%s\", \"type\": \"%s\", \"properties\": %s}",
+                message.getName(), message.getType(), properties);
+    }
+
+    protected String getPropertiesWithBlob(DocumentMessage message) {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode node = mapper.valueToTree(message.getProperties());
+        ObjectNode fileContent = mapper.createObjectNode();
+        // use a place holder for the batch id
+        fileContent.put("upload-batch", BATCH_ID_TAG);
+        fileContent.put("upload-fileId", "0");
+        return ((ObjectNode) node).set("file:content", fileContent).toString();
+    }
+
+    protected String getProperties(DocumentMessage message) throws JsonProcessingException {
+        return new ObjectMapper().writeValueAsString(message.getProperties());
+    }
+
+    protected byte[] bytes(String val) {
+        try {
+            return val.getBytes("UTF-8");
```

Comment:
Use a static import of `StandardCharsets.UTF_8` instead of `"UTF-8"` (in several places)